_source/_posts/2020-10-22-set-up-a-mute-indicator-light-for-zoom-with-hammerspoon.md

```diff
@@ -0,0 +1,283 @@
+
+Once the `~/.hammerspoon/init.lua` configuration file is open, use your favorite text editor to add the following line to the file:
+
+```lua
+hs.loadSpoon("Zoom")
+```
+
+Save the file, then click on the Hammerspoon menu icon, and select **Reload config**, then click on the the Hammerspoon menu icon again to select **Console**.
+
+At this point, should see the Hammerspoon console, and it should look like this:
+
+{% img blog/zoom-hammerspoon/console-zoom-inmeeting.png alt:"Image of the Hammerspoon preferences window" width:"512px" %}{: .center-image }
+
+Now, at first glance, this console appears to be just a message log. Not so! You can also interactively run Lua in this console, which is exactly what we're going to do now.
+
+To test out the unofficial Zoom Spoon, run this command in the Hammerspoon console:
+
+```lua
+spoon.Zoom:inMeeting()
+```
+
+If Zoom is already open, and in a meeting, this command should return `true` otherwise, it'll return `false`
+
+{% img blog/zoom-hammerspoon/console-zoom-inmeeting.png alt:"Image of the Hammerspoon preferences window" width:"512px" %}{: .center-image }
+
+If the command returned `false` like in the image above, then open Zoom, start a meeting, and re-run the command. This time around, it should return `true`
+
+{% img blog/zoom-hammerspoon/console-zoom-inmeeting.png alt:"Image of the Hammerspoon preferences window" width:"512px" %}{: .center-image }
+
+
+If this doesn't work, it probably means that you need to give Hammerspoon permission to control your computer. To do this, open System Preferences, and go to **Security & Privacy**>**Privacy**" >**Accessibility**, unlock the preference pane to make changes, then select the checkbox next to "Hammerspoon" to allow Hammerspoon to inspect the state of Zoom.
+
+{% img blog/zoom-hammerspoon/console-zoom-inmeeting.png alt:"Image of the Hammerspoon preferences window" width:"512px" em%}{: .center-image }
+
+
+Once you get the `spoon.Zoom:inMeeting()` command working, two others for you to try are `spoon.Zoom:mute()` and `spoon.Zoom:unmute()` try out both while you have a live Zoom meeting running and see what happens!
+
+
+## Configure Hammerspoon to use the Zoom Spoon
+
+At this point, you should be able to run the `spoon.Zoom:mute()` and `spoon.Zoom:unmute()` commands to mute and unmute a Zoom meeting. However, this isn't really any more useful than what you can already do with Zoom, so in this section I'll be showing you how to set up a mute indicator in your menu bar.
+
+**Note:** Zoom provides a global mute and unmute hotkey out of the box! If all you care about is a global mute and unmute hotkey, I suggest using that!
+
+Because I assume that most people don't have a blink() USB LED already, I'm going to show you how to do the next best thing which is to simulate this LED in your menu bar.
```

Comment:
This website seems to indicate it should be blink(1) instead of blink().

https://blink1.thingm.com/